packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1230,6 +1243,53 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     });
   }
 
+  /// Whether this has been disposed.
+  ///
+  /// If asserts are disabled, this property is always null.
+  bool? get debugDisposed {
+    bool? disposed;
+    assert(() {
+      disposed = _debugDisposed;
+      return true;
+    }());
+    return disposed;
+  }
+
+  bool _debugDisposed = false;
+
+  /// Release any resources held by this render object.
+  ///
+  /// The object that creates a RenderObject is in charge of disposing it.
+  /// If this render object has created any children directly, it must dispose
+  /// of those children in this method as well. It must not dispose of any
+  /// children that were created by some other object, such as
+  /// a [RenderObjectElement]. Those children will be disposed when that
+  /// element unmounts, which may be delayed if the element is moved to another
+  /// part of the tree.
+  ///
+  /// If [isRepaintBoundary] returns true, the layer tree rooted at this
+  /// object's layer will also dispose.
+  ///
+  /// Implementations of this method must end with a call to the inherited
+  /// method, as in `super.dispose()`.
+  ///
+  /// The object is no longer usable after calling dispose.
+  @mustCallSuper
+  void dispose() {
+    assert(!_debugDisposed);
+    _layer = null;
+    assert(() {
+      visitChildren((RenderObject child) {
+        assert(
+          child.debugDisposed!,
+          '${child.runtimeType} (child of $runtimeType) must be disposed before calling super.dispose().',
```

Comment:
Maybe add a note about this condition to the doc comment?

Reply:
This corresponds to the doc comment 

```
  /// Implementations of this method must end with a call to the inherited
  /// method, as in `super.dispose()`.
```